design/src/rfcs/rfc0037_http_wrapper.md

```diff
@@ -0,0 +1,96 @@
+<!-- Give your RFC a descriptive name saying what it would accomplish or what feature it defines -->
+RFC: The HTTP Wrapper Type
+=============
+
+<!-- RFCs start with the "RFC" status and are then either "Implemented" or "Rejected".  -->
+> Status: RFC
+>
+> Applies to: client
+
+<!-- A great RFC will include a list of changes at the bottom so that the implementor can be sure they haven't missed anything -->
+For a summarized list of proposed changes, see the [Changes Checklist](#changes-checklist) section.
+
+<!-- Insert a short paragraph explaining, at a high level, what this RFC is for -->
+This RFC defines the API of our wrapper types around `http::Request` and `http::Response`. For more information about why we are wrapping these types, see [RFC 0036: The HTTP Dependency](./rfc0036_http_dep_elimination.md).
+
+<!-- The "Terminology" section is optional but is really useful for defining the technical terms you're using in the RFC -->
+Terminology
+-----------
+
+- **Some Term**: A definition for that term
+
+<!-- Explain how users will use this new feature and, if necessary, how this compares to the current user experience -->
+The user experience if this RFC is implemented
+----------------------------------------------
+
+In the current version of the SDK, external customers and internal code interacts directly with the [`http`](https://crates.io/crates/http) crate. Once this RFC is implemented, interactions at the **public** API level will occur with our own `http` types instead.
+
+Our types aim to be nearly drop-in-compatible for types in the `http` crate, however:
+1. We will not expose existing HTTP types in public APIs in ways that are ossified.
+2. When possible, we aim to simplify the APIs to make them easier to use.
+3. We will add SDK specific helper functionality when appropriate, e.g. first-level support for applying an endpoint to a request.
+
+<!-- Explain the implementation of this new feature -->
+How to actually implement this RFC
+----------------------------------
+
+We will need to add two types, `HttpRequest` and `HttpResponse`.
+
+#### To string or not to String
+The `http` library is very precise in its representation—specifically it allows for `HeaderValue`s that are both a super and subset of `String`—a superset because headers support arbitrary binary data but a subset because headers can contain any sequence of non-zero codepoints including sequences that are not valid UTF-8.
```

Comment:
I don't think this section actually answers the question of String or not String